dev/devicelab/lib/framework/adb.dart

```diff
@@ -391,16 +391,20 @@ class FuchsiaDeviceDiscovery implements DeviceDiscovery {
     final Map<String, HealthCheckResult> results = <String, HealthCheckResult>{};
     for (final String deviceId in await discoverDevices()) {
       try {
+        StringBuffer stderr;
         final int resolveResult = await exec(
-          _devFinder,
+          _ffx,
           <String>[
-            'resolve',
-            '-device-limit',
-            '1',
+            'target',
+            'list',
+            '--format',
+            'a',
             deviceId,
-          ]
+          ],
+          stderr: stderr
         );
-        if (resolveResult == 0) {
+        final String stderrOutput = stderr.toString().trim();
+        if (resolveResult == 0 && ! stderrOutput.contains('No devices found')) {
```

Comment:
Is this baked into the public API contract? Is there a better mechanism that we can use here to try to parse the output here?

Reply:
Currently no, there are talks going on about defining it.